client-common/src/storage.rs

```diff
@@ -242,37 +232,24 @@ pub fn encrypt_bytes<K: AsRef<[u8]>>(
     Ok(cipher)
 }
 
-/// Decrypts bytes with given passphrase
-pub fn decrypt_bytes<K: AsRef<[u8]>>(
-    key: K,
-    passphrase: &SecUtf8,
-    bytes: &[u8],
-) -> Result<Vec<u8>> {
-    let nonce = &bytes[..NONCE_SIZE];
-    let salt = &bytes[NONCE_SIZE..NONCE_SIZE + SALT_SIZE];
-
-    let payload = &bytes[NONCE_SIZE + SALT_SIZE..];
-
-    let algo = get_algo(passphrase, salt)?;
+/// Decrypts bytes with given enckey
+pub fn decrypt_bytes<K: AsRef<[u8]>>(key: K, enckey: &SecKey, bytes: &[u8]) -> Result<Vec<u8>> {
+    let algo = get_algo(enckey)?;
 
     let payload = Payload {
-        msg: payload,
+        msg: &bytes[NONCE_SIZE..],
         aad: key.as_ref(),
     };
 
-    algo.decrypt(GenericArray::from_slice(nonce), payload)
+    algo.decrypt(GenericArray::from_slice(&bytes[..NONCE_SIZE]), payload)
         .map_err(|_| {
             Error::new(
                 ErrorKind::DecryptionError,
-                "Incorrect passphrase: Unable to unlock stored values",
+                "Incorrect enckey: Unable to unlock stored values",
             )
         })
 }
 
-fn get_algo(passphrase: &SecUtf8, salt: &[u8]) -> Result<Aes256GcmSiv> {
-    let hash = hash_raw(passphrase.unsecure().as_bytes(), salt, &Default::default())
-        .chain(|| (ErrorKind::HashError, "Unable to hash passphrase"))?;
-
-    let key = GenericArray::clone_from_slice(hash.as_ref());
-    Ok(Aes256GcmSiv::new(key))
+fn get_algo(enckey: &SecKey) -> Result<Aes256GcmSiv> {
```

Comment:
Not return a `Result` from this function?

Reply:
Done